subsys/bluetooth/shell/bt.c

```diff
@@ -193,26 +193,37 @@ static bool data_cb(struct bt_data *data, void *user_data)
 	}
 }
 
-static void print_data_set(uint8_t type, uint8_t set_value_len,
+static void print_data_hex(const uint8_t *data, uint8_t len, bool reverse)
+{
+	shell_fprintf(ctx_shell, SHELL_INFO, "0x");
+	for (uint8_t i = 0; i < len; i++) {
+		shell_fprintf(ctx_shell, SHELL_INFO, "%02x", data[reverse ? len-1-i : i]);
+	}
+}
+
+static void print_data_set(uint8_t type, uint8_t initial_value_len, uint8_t set_value_len,
```

Comment:
Have you considered calling this function two times, instead of adding this additional `set_value_len` handling? 

```c
	case BT_DATA_SVC_DATA16:
		print_data_set(data->type, 2, 1, data->data, data->data_len);
		break;
```
would be 
```c
	case BT_DATA_SVC_DATA16:
		print_data_set(data->type, 2, data->data, data->data_len);
		print_data_set(data->type, 1, data->data + 2, data->data_len - 2);
		break;
```

The issue with this approach is that is assumes that all data after the initial value is the same size, which may be true for the few specific AD type handled so far, but is definitely not an assumption that will generally hold.

Reply:
I am ok with a two call approach.  However, we also need to include how many times it should be printed - unless we want to do the loop in the switch block.  I can include a count arg.